test/validator/dependency_test.dart

```diff
@@ -56,213 +65,194 @@ Future<void> setUpDependency(dep,
+        ).create();
 
-      await expectValidation();
-    });
+        await expectValidation(
+          environment: {
+            '_PUB_TEST_SDK_VERSION': '1.19.0',
+            'FLUTTER_ROOT': path.join(d.sandbox, 'flutter'),
+          },
+        );
+      },
+    );
 
     test(
-        'depends on a package from Fuchsia with an appropriate Dart SDK '
-        'constraint', () async {
-      await package(sdk: '>=2.0.0-dev.51.0 <2.0.0', deps: {
-        'foo': {'sdk': 'fuchsia', 'version': '>=1.2.3 <2.0.0'}
-      }).create();
-      await d.validPackage.create();
+      'depends on a package from Fuchsia with an appropriate Dart SDK constraint',
+      () async {
+        await fuschiaPackage('foo', sdk: '>=2.0.0 <3.0.0').create();
+        await package(sdk: '>=2.0.0 <3.0.0', deps: {
+          'foo': {'sdk': 'fuchsia', 'version': '>=1.2.3 <2.0.0'}
+        }).create();
 
-      await expectValidation();
-    });
+        await expectValidation(environment: {
+          'FUCHSIA_DART_SDK_ROOT': path.join(d.sandbox, 'fuchsia'),
+          '_PUB_TEST_SDK_VERSION': '2.12.0',
+        });
+      },
+    );
   });
 
   group('should consider a package invalid if it', () {
     setUp(package().create);
-    group('has a git dependency', () {
-      group('where a hosted version exists', () {
-        test('and should suggest the hosted primary version', () async {
-          await setUpDependency({'git': 'git://github.com/dart-lang/foo'},
-              hostedVersions: ['3.0.0-pre', '2.0.0', '1.0.0']);
-          await expectValidationWarning('  foo: ^2.0.0');
-        });
-
-        test(
-            'and should suggest the hosted prerelease version if '
-            "it's the only version available", () async {
-          await setUpDependency({'git': 'git://github.com/dart-lang/foo'},
-              hostedVersions: ['3.0.0-pre', '2.0.0-pre']);
-          await expectValidationWarning('  foo: ^3.0.0-pre');
-        });
-
-        test(
-            'and should suggest a tighter constraint if primary is '
-            'pre-1.0.0', () async {
-          await setUpDependency({'git': 'git://github.com/dart-lang/foo'},
-              hostedVersions: ['0.0.1', '0.0.2']);
-          await expectValidationWarning('  foo: ^0.0.2');
-        });
-      });
-
-      group('where no hosted version exists', () {
-        test("and should use the other source's version", () async {
-          await setUpDependency({
-            'git': 'git://github.com/dart-lang/foo',
-            'version': '>=1.0.0 <2.0.0'
-          });
-          await expectValidationWarning('  foo: ">=1.0.0 <2.0.0"');
-        });
-
-        test(
-            "and should use the other source's unquoted version if "
-            'concrete', () async {
-          await setUpDependency(
-              {'git': 'git://github.com/dart-lang/foo', 'version': '0.2.3'});
-          await expectValidationWarning('  foo: 0.2.3');
-        });
-      });
-    });
```

Comment:
I'm not sure I follow why these where valid before, and why we're removing them now.
Just curious?

Reply:
Thanks - they should not have been removed.

Reinstated them.

We might consider how useful these checks are: https://github.com/dart-lang/pub/issues/3567